storage/replica_command.go

```diff
@@ -186,6 +178,16 @@ func (r *Replica) executeCmd(
 	header.ResumeSpan = span
 	reply.SetHeader(header)
 
+	// TODO(peter): We'd like to assert that the hlc clock is always updated
+	// correctly, but various tests insert versioned data without going through
+	// the proper channels. See TestPushTxnUpgradeExistingTxn for an example.
+	//
+	// if header.Txn != nil && !header.Txn.Timestamp.Less(h.Timestamp) {
+	// 	if now := r.store.Clock().Now(); now.Less(header.Txn.Timestamp) {
+	// 		log.Fatalf(ctx, "hlc clock not updated: %s < %s", now, header.Txn.Timestamp)
+	// 	}
+	// }
```

Comment:
@tschottdorf This doesn't look easy to fix. I think we shouldn't gate this change on doing so.